### ISSUE TITLE
Interpret time with just 3 digits

### ISSUE THREAD

**Issue by CarlosGarcell** [moment/moment#3269]
When attempting to convert a 3 digit string into a valid time string, moment throws an "Invalid date" error. Here is an example of what I'm seeing:

```
Input -> Return Value
1200 -> 12:00
0845 -> 08:45
845 -> Invalid date
```

Wouldn't it be accurate or safe to assume that, if the input consists of 3 digits with no leading '0', the output should be an appropriately formatted time (as long as that value is valid within the range of 00:00-23:59)? That way, 845 would come out as 08:45.

FYI, I'm on version 2.10.6 of the library.

**Comment by mj1856** [comment]
It works fine, as long as you specify the input format.

```javascript
moment("1200", "Hmm").format("HH:mm")  // "12:00"
moment("0845", "Hmm").format("HH:mm")  // "08:45"
moment("845", "Hmm").format("HH:mm")   // "08:45"
```

**Comment by CarlosGarcell** [comment]
It is definitely not working for me, @mj1856, I even copied the last line you sent me, just to make sure I did not have an error in my code, what would you advise me to do?

**Comment by icambron** [comment]
@CarlosGarcell Please post a JSFiddle demonstrating the issue. @ me here when you do it and I'll either help or reopen the issue

**Comment by CarlosGarcell** [comment]
Hello @icambron, after testing on JSFiddle with different versions, it turns out that the version we're currently using in our project is the one that would return an invalid date the input consists of 3 digits. These were the results

```
// Version 2.10.6
moment("845", "Hmm").format("HH:mm"); // Invalid date

// Version 2.11.0
moment("845", "Hmm").format("HH:mm"); // 08:45
```

I did include the version we're using in my original post to make sure you guys had all the information necessary to debug the issue.

Thank you for your help!

**Comment by icambron** [comment]
Great, thanks for checking. Looks like this was fixed in #2700.
